### PR TITLE
Clarify the process for changing the governance

### PR DESCRIPTION
**Pull request by jasongrout**
This clarifies the procedure for changing the governance, including submitting a pull request and calling for a formal steering council vote.

*Update: +1 from @jasongrout, @ellisonbg, @minrk, @damianavila, @Carreau, @rgbkrk, @jhamrick @fperez*

### CODE REVIEW

**Comment by jasongrout** [comment]
Brian mentions in https://github.com/jupyter/governance/pull/5#issuecomment-137819085

> In general, we haven't used formal votes with the project in the past. I
> think informal consensus is a much better model as it encourages us to find
> solutions that more people are happy with. Votes can lead to a culture
> where 49.99999% of the people are unhappy about each decision.

**Comment by jasongrout** [comment]
I would say the development of the PR should be the "informal consensus" stage in the development of the changes.  The vote should be more of a ratification than a battle.  Again, I'm proposing this just for changing the governance model, which I think should have the highest standards of formality for procedure.

Probably the bar should be higher than "majority".  BDFL authority also is there to make sure things don't get too out of hand.

**Comment by ellisonbg** [comment]
I like the idea of voting being just the ratification step after informal consensus has emerged.

Some options on what it takes to accept changes:

* Unanimous of all SC members
* 2/3 majority of all SC members
* 3/4 majority of all SC members

I think it is important to have these fractions relative to the full SC membership, not just those who vote. This insure that SC members note, but also clarifies when enough SC members have voted to allow things to move forward.

**Comment by Carreau** [comment]
Need rebase.

**Comment by jasongrout** [comment]
I'll rebase it once we have a good idea of what we want for the voting model.  @ellisonbg - another standard model is to have a minimum quorum that can transact business, and then it's majority, or super majority, etc., of the quorum.  So, for example, we could say that 80% have to vote to form a quorum, and it's a 2/3 majority vote.  With the current 10 members, that would mean at least 8 have to vote, and 2/3 majority would mean that 6 would need to vote positive.  I think this is better than saying just 6 positives pass, since it does make sure that at least 8 people participated in the vote.

I think the U.S. Congress has a quorum of "a majority", for example.

**Comment by damianavila** [comment]
The quorum idea is nice I think...

File: governance.md [hunk 1]
```diff
@@ -323,3 +323,19 @@ Existing **Tier 2** Partners:
 
 -   Bloomberg (Jason Grout, Sylvain Corlay)
 -   Google (Kester Tong, Kayur Patel, Mark Sandler)
+
+
+Changing the Governance
```
Comment (ellisonbg) [inline governance.md]: Maybe "Changing the"

* "Governance Model"
* "Governance Policy"
* "Governance Document"

?

File: governance.md [hunk 2]
```diff
@@ -323,3 +323,19 @@ Existing **Tier 2** Partners:
...
+=======================
+
+Changes to the governance documents are proposed via a GitHub pull
+request to The Project's governance documents GitHub repository at
+[https://github.com/jupyter/governance](https://github.com/jupyter/governance).
+After a period of public comment and review, a Steering Council Member
+proposes to the Steering Council that the pull request be merged
+(accepting the proposed changes) or proposes that the pull request be
+closed without merging (rejecting the proposed changes).  The Member
+should list the commit range proposed for acceptance or rejection and
```
Comment (ellisonbg) [inline governance.md]: I think that just pointing to the PR is fine (rather than the commit range).

File: governance.md [hunk 3]
```diff
@@ -323,3 +323,19 @@ Existing **Tier 2** Partners:
...
+briefly summarize the pull request.  A majority vote from the Steering
+Council is needed to carry out the action.  The BDFL has authority to
```
Comment (ellisonbg) [inline governance.md]: I like your idea of a 80% quorum with 2/3 of the voters required to pass (rounding to nearest ints)

Comment (ellisonbg) [inline governance.md]: I think we do need language here that expresses our desire and intention for the open, community based discussion to reach consensus before a vote is proposed.

File: governance.md (hunk 2 above)
Comment (jasongrout) [inline governance.md]: I can change it to the final commit.  Prs change over time.

**Comment by jasongrout** [comment]
I'll also add an explicit mention that the bdfl can do whatever he wants

File: governance.md (hunk 1 above)
Comment (damianavila) [inline governance.md]: Documents, I think...

File: governance.md (hunk 3 above)
Comment (damianavila) [inline governance.md]: > I like your idea of a 80% quorum with 2/3 of the voters required to pass (rounding to nearest ints)

Yep... I think this is important...

>  think we do need language here that expresses our desire and intention for the open, community based discussion to reach consensus before a vote is proposed.

Something along "all the community is encourage to participate actively in the discussion of these PRs..."

File: governance.md [hunk 4]
```diff
@@ -323,3 +323,19 @@ Existing **Tier 2** Partners:
...
+override the vote.
```
Comment (damianavila) [inline governance.md]: veto is the correct word for this action by the BDLF, but maybe it is less understandable than the override explanation...

**Comment by jasongrout** [comment]
I rebased, and I think I addressed each of the comments.

File: governance.md (hunk 2 above)
Comment (jasongrout) [inline governance.md]: (To clarify my brief comment) - I think in this particular formal process, we should be very explicit about what is being voted on.  I also think it's fine for the steering council members to say that their vote applies to the proposed commit, as well as grammar fixes in the proposed commit, like in your recent proposal (so members can make their vote more broad than the proposed commit, or they can just vote for the proposed commit.)

**Comment by damianavila** [comment]
LGTM

File: governance.md (hunk 4 above)
Comment (jasongrout) [inline governance.md]: Yes, I wanted to avoid the double negative confusion if Fernando vetoed a rejection.

Comment (damianavila) [inline governance.md]: Yep, I understand :+1:

Comment (minrk) [inline governance.md]: You didn't misunderstand :)

**Comment by Carreau** [comment]
+1

**Comment by rgbkrk** [comment]
Seems alright to me.

**Comment by jasongrout** [comment]
I've proposed that these changes be ratified by the steering council.

**Comment by rgbkrk** [comment]
:+1: again, :shipit:

**Comment by jasongrout** [comment]
+1 from me, of course.

**Comment by njsmith** [comment]
An interesting cautionary tale regarding voting rules: X.org is trying to dissolve their foundation and switch to being a sponsored project of SPI. The voting rule in their bylaws is that changes to the bylaws require 2/3 of foundation "members". The [vote failed](http://www.x.org/wiki/BoardOfDirectors/Elections/2015Results/) with 95% in favor because only 60% of their membership voted.

Very unlikely to be an issue with Jupyter's structure, but it's nice to have these kinds of examples in one's pocket, I think.

**Comment by damianavila** [comment]
+1 again...

**Comment by jasongrout** [comment]
Math

**Comment by jasongrout** [comment]
Nathaniel, thanks.  There are a lot more examples from the various congresses :).  Two mitigating factors in our case: the bdfl does whatever he wants anyway, and the active set of people voting is the steering council which is a relatively small set hand-picked for their activity--if they fail to vote too many times or participate, then their active steering council membership status is in question.

**Comment by ellisonbg** [comment]
+1

On Sat, Sep 19, 2015 at 12:37 AM, Jason Grout <notifications@github.com>
wrote:

> Nathaniel, thanks. There are a lot more examples from the various
> congresses :). Two mitigating factors in our case: the bdfl does whatever
> he wants anyway, and the active set of people voting is the steering
> council which is a relatively small set hand-picked for their activity--if
> they fail to vote too many times or participate, then their active steering
> council membership status is in question.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/governance/pull/6#issuecomment-141633330>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by jasongrout** [comment]
@minrk also voted +1.

**Comment by fperez** [comment]
Yes, I think that the activity point is one we need to reiterate: we should continuously prune the council, not in any punitive way, but simply as a reflection that people get busy, are swamped and overwhelmed, have other priorities, etc. And we do *not* want to get deadlocked by artificial bureaucracy.

So, I'm +1 on this change, but I want to emphasize that the 'seat of power' is our community, *not* the council, nor me (I'm mostly there as a safety valve, not as a power mechanism). We want to *decentralize* as much as possible into working groups and other mechanisms that empower the community to run with new ideas and make them grow, without having to constantly come to the council for decisions and votes.  If nothing else, because we all know we will simply slow to a crawl on those decisions, or make them poorly, or rubber-stamp them.

In the spirit of involving the community, I'm not sure how aware folks on the main list are of this thread, perhaps it wouldn't hurt if @jasongrout who has been championing this gives a last ping in case anyone has additional thoughts.  I really want to make sure we maintain as much of our governance as humanly possible on our public and community channels...

**Comment by jasongrout** [comment]
@fperez: done: https://groups.google.com/forum/#!topic/jupyter/dEMEY7x2s8w

**Comment by fperez** [comment]
thx!

**Comment by jasongrout** [comment]
We have 8/8 positive responses from the steering council, and no other responses from the community.  I'll merge this.

**Comment by ellisonbg** [comment]
Great, many thanks Jason!

On Tue, Sep 22, 2015 at 8:32 AM, Jason Grout <notifications@github.com>
wrote:

> Merged #6 <https://github.com/jupyter/governance/pull/6>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/governance/pull/6#event-415990282>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by jdfreder** [comment]
A late +1 : )

**Pull request merged**